js/views/pageNavVw.js

```diff
@@ -356,6 +358,27 @@ module.exports = baseVw.extend({
     app.hideOverlay();
     this.$notifMenu.removeClass('popMenu-opened');
   },
+  
+  markNotificationAsRead: function(view) {
```

Comment:
We are already marking the notifications as read when the menu is closed. Please see onNotifMenuClose(). We are also marking all unread notifications in the menu as read in one batch call, rather than making a separate call for each notification. We'd like the "mark as read" functionality to be centralized there, since there is a bit of complexity to keeping track of an accurate unread count.

Thanks for finding this bug. It's a legitimate one, but the crux of the problem is that onNotifMenuClose() is not being called when you click on a notification, even though the menu is visually closing. I have another count related bug I'll be working on shortly. I'll take a look into why the method is not being called.

Thanks for your help!